---
title: "Examine_nonoverlapping_STARF_Arriba"
author: "bhaas"
date: '2024-09-10'
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```


```{r}


data = read.csv("illum_fusions_w_gene_coords.tsv.wAnnots", header=T, sep="\t")

data %>% head()


```

```{r}
# restrict to those fusion gene pairs that have long read support.

fusions_w_long_reads = data %>% filter(has_long_read_support) %>% select(lex_ordered_fusion_name) %>% unique() %>% pull(lex_ordered_fusion_name)

data = data %>% filter(lex_ordered_fusion_name %in% fusions_w_long_reads)

```




```{r}

data = data %>% mutate(sum_arriba_reads = split_reads1 + split_reads2 + discordant_mates) %>%
    mutate(sum_starF_reads = est_J + est_S) 


data %>% head()

```

```{r}

# sum across fusion isoforms per gene
summary = data %>% group_by(lex_ordered_fusion_name) %>% summarize(StarF_sum_reads = sum(sum_starF_reads, na.rm = T), 
                                                             Arriba_sum_reads = sum(sum_arriba_reads, na.rm=T) )

summary %>% head()

```




```{r}


# compare counts of evidence reads

summary %>% ggplot(aes(y=StarF_sum_reads, x=Arriba_sum_reads)) + geom_point() + xlim(0,250) + ylim(0,250) +
    theme_bw() +
    xlab("Arriba split read count") +
    ylab("StarF split read count") +
    geom_abline(slope=1, intercept=0)
    


```


```{r}

cor(summary$StarF_sum_reads, summary$Arriba_sum_reads, use='complete.obs')


```


```{r}



summary = summary %>% mutate(progs = 'both') %>%
    mutate(progs = ifelse(StarF_sum_reads == 0, 'arriba-only', progs)) %>%
    mutate(progs = ifelse(Arriba_sum_reads  == 0, 'starF-only', progs)) 

summary %>% head()


```

```{r}

summary = summary %>% mutate(sum_reads = ifelse( (StarF_sum_reads > 0 & Arriba_sum_reads > 0 ), 
                                                 (StarF_sum_reads + Arriba_sum_reads/2),  
                                                 StarF_sum_reads)) %>%
    mutate(sum_reads = ifelse(sum_reads == 0, 
                              Arriba_sum_reads, 
                              sum_reads) )

summary %>% head()
```



```{r}

summary %>% ggplot(aes(x = progs, y=sum_reads, color=progs)) + geom_jitter() +
    theme_bw() +
    ylim(0,250) +
    ggtitle("Compare read support for fusion pairs according to method")

```



# compare both vs. arriba-only

```{r}

wilcox.test(summary %>% filter(progs == "both") %>% pull(sum_reads),
            summary %>% filter(progs == "arriba-only") %>% pull(sum_reads), alternative = 'greater')


```



# compare both vs. starF-only


```{r}

wilcox.test(summary %>% filter(progs == "both") %>% pull(sum_reads),
            summary %>% filter(progs == "starF-only") %>% pull(sum_reads), alternative = 'greater')

```






